sample/src/main/java/com/uber/autodispose/recipes/AutoDisposeFragment.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2017. Uber Technologies
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.autodispose.recipes;
+
+import android.app.Fragment;
+import android.content.Context;
+import android.os.Bundle;
+import android.view.View;
+import com.uber.autodispose.LifecycleEndedException;
+import com.uber.autodispose.LifecycleScopeProvider;
+import io.reactivex.Observable;
+import io.reactivex.functions.Function;
+import io.reactivex.subjects.BehaviorSubject;
+import javax.annotation.Nullable;
+
+/**
+ * A {@link Fragment} example implementation for making one implement {@link
+ * LifecycleScopeProvider}. One would normally use this as a base fragment class to extend others
+ * from.
+ */
+public abstract class AutoDisposeFragment extends Fragment
+    implements LifecycleScopeProvider<AutoDisposeFragment.FragmentEvent> {
+
+  public enum FragmentEvent {
+    ATTACH, CREATE, CREATE_VIEW, START, RESUME, PAUSE, STOP, DESTROY_VIEW, DESTROY, DETACH
```

Comment:
The same issue stays here — for fragments it's even harder to well-understand this small FSM mechanism.

Reply:
what does FSM mechanism mean? 👀